packages/cubejs-schema-compiler/src/adapter/PreAggregations.js

```diff
@@ -362,82 +363,175 @@ export class PreAggregations {
     }
   }
 
+  /**
+   * Returns function to determine whether pre-aggregation can be used or not
+   * for specified query, or its value for `refs` if specified.
+   * @param {Object} transformedQuery transformed query
+   * @param {Object?} refs pre-aggs reference
+   * @returns {function(preagg: Object): boolean}
+   */
   static canUsePreAggregationForTransformedQueryFn(transformedQuery, refs) {
+    /**
+     * Returns an array of 2-elements arrays with the dimension and granularity
+     * sorted by the concatenated dimension + granularity key.
+     * @param {Array<{dimension: string, granularity: string}>} timeDimensions
+     * @returns {Array<Array<string>>}
+     */
+    const sortTimeDimensions = (timeDimensions) => (
+      timeDimensions &&
+      R.sortBy(
+        d => d.join('.'),
```

Comment:
There are a number of codecov warnings along the lines of `Added line #L383 was not covered by tests`. These can be potential sources of bugs, especially since this is vanilla JS and there is little typing guardrails. Consider adding some tests to increase test coverage.

Reply:
Here we have unit tests and a bunch of integration tests, which, unfortunately, are not analyzed by the coverage engine. Generally speaking, this area is tested very well. We can improve coverage, for sure, but in the scope of some other tasks))